tensorflow_model_optimization/python/core/sparsity/keras/pruning_impl.py

```diff
@@ -28,7 +29,7 @@ class Pruning(object):
   """Implementation of magnitude-based weight pruning."""
 
   def __init__(self, training_step_fn, pruning_vars, pruning_schedule,
-               block_size, block_pooling_type):
+               block_size, sparsity_2x4, block_pooling_type):
```

Comment:
Since this is not scalable when we want to support other types of structured sparsity, we eventually want to generalize this. 

Is there any reason you didn't generalize this to mxn sparsity as you did in pruning_utils.py ?

Reply:
we have not done this as we are interested in this case only and do all testing for it.
but yes, we can do generalized version.

Reply:
thanks for the explanation. It looks better to generalize it (make configurable API for other structures) as @liyunlu0618  mentioned below. It would be OK to test mainly on 2:4, since it is the interesting use case for now.

Reply:
Hi @rino20 I pushed the latest version for the review.